corehq/apps/userreports/specs.py

```diff
@@ -17,11 +22,54 @@ def TypeProperty(value):
 @dataclass
 class FactoryContext():
     named_expressions: dict
+    _named_expressions: dict = field(init=False, repr=False)
+
     named_filters: dict
+    _named_filters: dict = field(init=False, repr=False)
+
+    domain: Optional[str] = None
+
+    @property
+    @memoized
+    def named_filters(self):
+        extra_filters = {}
+        if self.domain and toggles.UCR_EXPRESSION_REGISTRY.enabled(self.domain):
+            from corehq.apps.userreports.models import UCRExpression
+            extra_filters = {
+                f.name: f.wrapped_definition(self.domain)
+                for f in UCRExpression.objects.filter(
+                    domain=self.domain,
+                    expression_type=UCR_NAMED_FILTER,
+                )
+            }
```

Comment:
nit: make this a manager method on the UCRExpression manager:
```
extra_filters = UCRExpression.objects.get_filters_for_domain(domain)
```

Reply:
https://github.com/dimagi/commcare-hq/pull/31443/commits/82f9d8ed9ffff2bd4b882e51a6a95c2280b1b2db